src/Symfony/Component/Translation/Tests/DependencyInjection/TranslationPassTest.php

```diff
@@ -20,31 +20,88 @@
 
 class TranslationPassTest extends TestCase
 {
+    /**
+     * @group legacy
+     *
+     * A test that verifies the "translation.reader" gets the LoaderInterfaces added.
+     *
+     * This test should be kept in 4.0 but without the legacy group.
+     */
     public function testValidCollector()
     {
         $loader = (new Definition())
             ->addTag('translation.loader', array('alias' => 'xliff', 'legacy-alias' => 'xlf'));
 
+        $reader = new Definition();
+
+        $translator = (new Definition())
+            ->setArguments(array(null, null, null, null));
+
+        $container = new ContainerBuilder();
+        $container->setDefinition('translator.default', $translator);
+        $container->setDefinition('translation.reader', $reader);
+        $container->setDefinition('translation.xliff_loader', $loader);
+
+        $pass = new TranslatorPass();
```

Comment:
right! Could we test the duplicated code in the legacy test below? It would ensure that both BC and FC layers are ok and would just be removed along with the layers in 4.0.